integrations/anthropic/src/ucai_anthropic/utils.py

```diff
@@ -0,0 +1,146 @@
+
+class ToolCallData:
+    """
+    Class representing a tool call, encapsulating the function name, arguments, and tool use ID.
+    """
+    def __init__(self, function_name: str, arguments: Dict[str, Any], tool_use_id: str):
+        self.function_name = function_name
+        self.arguments = arguments
+        self.tool_use_id = tool_use_id
+    
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts the ToolCallData instance into a dictionary.
+
+        Returns:
+            Dict[str, Any]: A dictionary representing the ToolCallData instance.
+        """
+        return {
+            "function_name": self.function_name,
+            "arguments": self.arguments,
+            "tool_use_id": self.tool_use_id
+        }
+
+    def execute(self, client: BaseFunctionClient) -> str:
+        """
+        Executes the Unity Catalog function using the provided client.
+
+        Args:
+            client (BaseFunctionClient): The client to use for executing the function.
+
+        Returns:
+            str: The result of the function execution.
+        """
+        result = client.execute_function(self.function_name, self.arguments)
+        return str(result.value)
+    
+    def to_tool_result_message(self, result: str) -> Dict[str, Any]:
+        """
+        Creates a tool result message based on the result of the function execution.
+
+        Args:
+            result (str): The result of the tool execution.
+
+        Returns:
+            Dict[str, Any]: A dictionary representing the tool result message.
+        """
+        return {
+            "type": "tool_result",
+            "tool_use_id": self.tool_use_id,
+            "content": result
+        }
+
+def extract_tool_call_data(response: Message) -> List[ToolCallData]:
+    """
+    Extracts the tool call information from the Anthropic response.
+
+    Args:
+        response (Message): The response from Anthropic containing tool usage blocks.
+
+    Returns:
+        List[ToolCallData]: A list of ToolCallData objects containing function names, arguments, and tool use IDs.
+    """
+    tool_calls_data = []
+
+    if response.stop_reason == "tool_use":
+        tool_calls = [tool_use for tool_use in response.content if isinstance(tool_use, ToolUseBlock)]
+        if tool_calls:
+            for tool_call in tool_calls:
+                if not tool_call.id:
+                    raise ValueError("Tool use block is missing an ID and is unable to resolve the tool call.")
+                
+                arguments = tool_call.input
+                func_name = construct_original_function_name(tool_call.name)
+                
+                tool_calls_data.append(ToolCallData(
+                    function_name=func_name,
+                    arguments=arguments,
+                    tool_use_id=tool_call.id
+                ))
+    
+    return tool_calls_data
+
+def generate_tool_call_messages(
+    *,
+    response: Message,
+    conversation_history: Any,
```

Comment:
Can we add types that the conversation_history should be? From the implementation it seems like List[Dict]?

Reply:
Added!